### PR TITLE
Use github action for FreeBSD testing and remove cirrus CI

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -521,6 +521,35 @@ jobs:
...
+    - name: "`make build`"
+      shell: bash
+      run: |
+        source $HOME/.cargo/env
+        cargo build
+    - name: "`make test`"
+      shell: bash
+      run: |
+        source $HOME/.cargo/env
+        cargo test -p uucore -p coreutils
```
Comment (miDeb) [inline .github/workflows/CICD.yml:551]: We should add `--features feat_os_unix` so that unix-specific utilities are tested as well

**Review by miDeb — CHANGES_REQUESTED** [review]

File: .github/workflows/CICD.yml (hunk 1 above)
Comment (sylvestre) [reply .github/workflows/CICD.yml]: yeah, indeed! I replicated what we had in cirrus. I am worried that it will surface some other issues but fine :)

**Review by tertsdiepraam — APPROVED** [review]
Looks good! This will also solve some issues where Cirrus was not merging the code with master before testing, leading to some unexpected results in the tests.

**Comment by miDeb** [comment]
It's weird that `test_users::test_users_check_name` (which is failing) is even running. It is marked with `#[cfg(any(target_vendor = "apple", target_os = "linux"))]`, so maybe `target_vendor` is the wrong thing to check for? Maybe we should check for `target_os = "macos"` instead? However, there are many places in the codebase where we check for `target_vendor = "apple"`.

**Comment by tertsdiepraam** [comment]
Could that have anything to do with the fact that it is running (in a FreeBSD vm) on macos? I would expect that should be unrelated though

**Comment by sylvestre** [comment]
Actually, the cargo tasks were running on the host :p 
All commands have to run into the freebsd task

**Comment by sylvestre** [comment]
A few failures:
```
failures:
    test_chmod::test_permission_denied
    test_chown::test_chown_only_group_id
    test_chown::test_chown_only_owner
    test_chown::test_chown_only_owner_colon
    test_chown::test_chown_only_user_id
    test_chown::test_chown_owner_group
    test_chown::test_chown_owner_group_id
    test_chown::test_chown_owner_group_mix
    test_chroot::test_enter_chroot_fails
    test_du::test_du_no_permission
    test_hostname::test_hostname_ip
    test_install::test_install_and_strip
    test_install::test_install_and_strip_with_program
    test_nice::test_negative_adjustment
    test_touch::test_touch_no_dereference
    test_touch::test_touch_permission_denied_error_msg
    test_touch::test_touch_system_fails
    test_uptime::test_uptime
    test_uptime::test_uptime_since
```

**Comment by miDeb** [comment]
Maybe some of the errors are because the test is run as `root`?

File: .github/workflows/CICD.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ name: CICD
 # spell-checker:ignore (env/flags) Awarnings Ccodegen Coverflow Cpanic RUSTDOCFLAGS RUSTFLAGS Zpanic
 # spell-checker:ignore (jargon) SHAs deps softprops toolchain
 # spell-checker:ignore (names) CodeCOV MacOS MinGW Peltoche rivy
-# spell-checker:ignore (shell/tools) choco clippy dmake dpkg esac fakeroot gmake grcov halium lcov libssl mkdir popd printf pushd rustc rustfmt rustup shopt xargs
+# spell-checker:ignore (shell/tools) choco clippy dmake dpkg esac fafkeroot gmake grcov halium lcov libssl mkdir popd printf pushd rustc rustfmt rustup shopt xargs
```
Comment (miDeb) [inline .github/workflows/CICD.yml:7]: I guess this change was unintentional?

File: .github/workflows/CICD.yml [hunk 3]
```diff
@@ -521,6 +521,49 @@ jobs:
         n_fails=$(echo "$output" | grep "^FAIL:\s" | wc --lines)
         if [ $n_fails -gt 0 ] ; then echo "::warning ::${n_fails}+ test failures" ; fi
 
+  test_freebsd:
+    runs-on: macos-latest
+    name: Tests/FreeBSD test suite
+    env:
+      # https://github.com/rust-lang/rustup/issues/2774
```
Comment (miDeb) [inline .github/workflows/CICD.yml:528]: This issue was closed, so maybe we don't need this anymore?

File: .github/workflows/CICD.yml (hunk 2 above)
Comment (sylvestre) [reply .github/workflows/CICD.yml]: yeah

**Comment by sylvestre** [comment]
Piouf, it wasn't smooth

Running as a user:
```
2021-08-23T21:17:22.7728560Z failures:
2021-08-23T21:17:22.7728980Z     test_chown::test_chown_only_group
2021-08-23T21:17:22.7729470Z     test_chown::test_chown_only_group_id
2021-08-23T21:17:22.7729960Z     test_chown::test_chown_owner_group
2021-08-23T21:17:22.7730470Z     test_chown::test_chown_owner_group_mix
2021-08-23T21:17:22.7730980Z     test_hostname::test_hostname_ip
2021-08-23T21:17:22.7731480Z     test_install::test_install_and_strip
2021-08-23T21:17:22.7732040Z     test_install::test_install_and_strip_with_program
2021-08-23T21:17:22.7732560Z     test_test::test_file_is_sticky
2021-08-23T21:17:22.7733060Z     test_touch::test_touch_no_dereference
2021-08-23T21:17:22.7733560Z     test_tr::test_more_than_2_sets
2021-08-23T21:17:22.7733990Z     test_tty::test_stdout_fail
```

**Comment by miDeb** [comment]
I know that other tests do this as well, but is there a reason why there is a build _and_ a test step? The test step does a build anyways.

**Comment by miDeb** [comment]
I think you accidentally reverted back to running the tests on the host machine

**Comment by sylvestre** [comment]
> I know that other tests do this as well, but is there a reason why there is a build _and_ a test step? The test step does a build anyways.

Good question, I am copying the legacy.
I guess we are doing that because tests are including more crates. Having this help to identify such issues (but I haven't seen any?!)

**Comment by sylvestre** [comment]
Oups,
However, the results that I shared are still the same:

```
    test_chown::test_chown_only_group
    test_chown::test_chown_only_group_id
    test_chown::test_chown_owner_group
    test_chown::test_chown_owner_group_mix
    test_hostname::test_hostname_ip
    test_install::test_install_and_strip
    test_install::test_install_and_strip_with_program
    test_test::test_file_is_sticky
    test_touch::test_touch_no_dereference
    test_tty::test_stdout_fail
```

**Comment by sylvestre** [comment]
We are good now :)

**Review by miDeb — APPROVED** [review]
Awesome that you could make this work!

**Comment by sylvestre** [comment]
Thanks.

**Pull request merged**